en/views/helpers/form.rst

```diff
@@ -535,334 +681,432 @@ applied to the generated HTML input element.
+    label caption equal to the string ``'empty'``. If you want to control
+    the label caption set this option to a string instead.
 
-.. code-block:: html
+  * When passed to a ``select`` method, this creates a blank HTML ``option``
+    element with an empty value in your drop down list. If you want to have an
+    empty value with text displayed instead of just a blank ``option``, pass a
+    string to ``'empty'``:
 
-    <textarea name="notes" id="notes">
-    This text is to be edited.
-    </textarea>
+      ::
 
-.. note::
+        echo $this->Form->select(
+            'field',
+            [1, 2, 3, 4, 5],
+            ['empty' => '(choose one)']
+        );
 
-    The ``textarea`` input type allows for the ``$options`` attribute
-    of ``'escape'`` which determines whether or not the contents of the
-    textarea should be escaped. Defaults to ``true``.
+      Output:
 
-::
+      .. code-block:: html
 
-    echo $this->Form->textarea('notes', ['escape' => false]);
-    // OR....
-    echo $this->Form->input('notes', ['type' => 'textarea', 'escape' => false]);
+          <select name="field">
+              <option value="">(choose one)</option>
+              <option value="0">1</option>
+              <option value="1">2</option>
+              <option value="2">3</option>
+              <option value="3">4</option>
+              <option value="4">5</option>
+          </select>
 
+* ``'hiddenField'`` - For checkboxes and radio buttons, by default,
+  a hidden ``input`` element is also created, along with the main
+  element, so that the key in ``$this->request->data``
+  will exist even without a value specified. For checkboxes its value
+  defaults to ``0`` and for radio buttons to ``''``.
 
-**Options**
+    Example of default output:
 
-In addition to the :ref:`general-input-options`, textarea() supports a few
-specific options:
+    .. code-block:: html
 
-* ``$options['rows'], $options['cols']`` These two keys specify the number of
-  rows and columns::
+        <input type="hidden" name="published" value="0" />
+        <input type="checkbox" name="published" value="1" />
 
-    echo $this->Form->textarea('textarea', ['rows' => '5', 'cols' => '5']);
+  This can be disabled by setting ``'hiddenField'`` to ``false``.
 
-  Output:
+    E.g. ::
 
-.. code-block:: html
+        echo $this->Form->checkbox('published', ['hiddenField' => false]);
 
-    <textarea name="textarea" cols="5" rows="5">
-    </textarea>
+    Which outputs:
+
+    .. code-block:: html
+
+        <input type="checkbox" name="published" value="1">
+
+  If you want to create multiple blocks of inputs on a form, that are
+  all grouped together, you should set this parameter to ``false`` on all
+  inputs except the first. If the hidden input is on the page in multiple
+  places, only the last group of inputs' values will be saved.
+
+    In this example, only the tertiary colors would be passed, and the
+    primary colors would be overridden:
```

Comment:
This is indented because I've indented by one extra level in all options enumeration lists, the code examples along with their introductory phrases (be they short, such as: "E.g.", "For example" or longer such as the one you indicated here).

The rationale for this was that I found  that some of the option enumeration lists lists (such as Options for Input, Common Options for Specific Input Types as well as Common Options for Select, Checkbox etc.) tend to occupy quite a bit of space and it becomes rather hard to differentiate between what's description/definition and what's example. I found that when landing in the middle of an options list it seems to be easier to visually scan through the list and "pick up" the definitions while ignoring the example if we're back just for a quick check (not a first read). I've fixed now the indenting to always  be 4 spaces per indentation level.
Of course it might be only my subjective impression, and if you think they should stay at the same level of indentation I'll change back the indent for all code/output examples along with their introductory phrase.
Just let me know.